library/common/event/provisional_dispatcher.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include "envoy/buffer/buffer.h"
+#include "envoy/event/deferred_deletable.h"
+#include "envoy/event/dispatcher.h"
+#include "envoy/http/api_listener.h"
+
+#include "common/common/logger.h"
+#include "common/common/thread.h"
+#include "common/common/thread_synchronizer.h"
+
+#include "absl/container/flat_hash_map.h"
+#include "absl/types/optional.h"
+#include "library/common/types/c_types.h"
+
+namespace Envoy {
+namespace Event {
+
+/**
+ * Wrapper around Envoy's Event::Dispatcher that queues functors until the underlying
+ * Event::Dispatcher has started.
+ */
+class ProvisionalDispatcher : public Logger::Loggable<Logger::Id::http> {
```

Comment:
Honestly, I spent forever thinking of a name, and this was the best I came up with. Open to suggestions though.

Provisional governments can become real governments once they ratify a constitution, etc.

But here's some other possibilities:

TransitionalDispatcher ("transition" is problematic because it's overloaded with other programming concepts)
InterimDispatcher (I like this better than "transitional", but it doesn't necessarily carry the connotation of something that can be permanent)
ProxyDispatcher (doesn't really capture any notion of serving a purpose between states; "proxy" is an overloaded term for networking)
QueueingDispatcher (but the real dispatcher queues, too)
DispatcherWrapper (sure, but why?)

Reply:
And I tried on like a hundred other names for size. ¯\\\_(ツ)\_/¯